pulsar-client-api/src/main/java/org/apache/pulsar/client/api/PulsarClientException.java

```diff
@@ -959,177 +960,55 @@ public TransactionHasOperationFailedException(String msg) {
-            return new BrokerPersistenceException(msg);
-        } else if (t instanceof BrokerMetadataException) {
-            return new BrokerMetadataException(msg);
-        } else if (t instanceof ProducerBusyException) {
-            return new ProducerBusyException(msg);
-        } else if (t instanceof ConsumerBusyException) {
-            return new ConsumerBusyException(msg);
-        } else if (t instanceof NotConnectedException) {
-            return new NotConnectedException();
-        } else if (t instanceof InvalidMessageException) {
-            return new InvalidMessageException(msg);
-        } else if (t instanceof InvalidTopicNameException) {
-            return new InvalidTopicNameException(msg);
-        } else if (t instanceof NotSupportedException) {
-            return new NotSupportedException(msg);
-        } else if (t instanceof NotAllowedException) {
-            return new NotAllowedException(msg);
-        } else if (t instanceof ProducerQueueIsFullError) {
-            return new ProducerQueueIsFullError(msg);
-        } else if (t instanceof ProducerBlockedQuotaExceededError) {
-            return new ProducerBlockedQuotaExceededError(msg);
-        } else if (t instanceof ProducerBlockedQuotaExceededException) {
-            return new ProducerBlockedQuotaExceededException(msg);
-        } else if (t instanceof ChecksumException) {
-            return new ChecksumException(msg);
-        } else if (t instanceof CryptoException) {
-            return new CryptoException(msg);
-        } else if (t instanceof ConsumerAssignException) {
-            return new ConsumerAssignException(msg);
-        } else if (t instanceof MessageAcknowledgeException) {
-            return new MessageAcknowledgeException(msg);
-        } else if (t instanceof TransactionConflictException) {
-            return new TransactionConflictException(msg);
-        } else if (t instanceof  TransactionHasOperationFailedException) {
-            return new TransactionHasOperationFailedException(msg);
-        } else if (t instanceof PulsarClientException) {
-            return new PulsarClientException(msg);
-        } else if (t instanceof CompletionException) {
+        if (t instanceof CompletionException) {
             return t;
-        } else if (t instanceof RuntimeException) {
-            return new RuntimeException(msg, t.getCause());
         } else if (t instanceof InterruptedException) {
             return t;
         } else if (t instanceof ExecutionException) {
             return t;
         }
 
-        return t;
+        try {
+            String combinedMessage = msg;
+            String originalMessage = t.getMessage();
+            if (originalMessage != null) {
+                combinedMessage += "\n" + originalMessage;
+            }
+            Constructor<? extends Throwable> constructor = t.getClass().getConstructor(String.class);
+            Throwable newException = constructor.newInstance(combinedMessage);
```

Comment:
Using reflection is a problem. It's better to remain using pure Java for this use case. Reflection will also cause issues with GraalVM native compilation.

Reply:
It might be possible to achieve a similar solution without reflection. I'd prefer instance methods in this case. The method names "wrap" and "unwrap" are not very intuitive, but it's impossible to rename the methods since that could possible break the API if these are considered to be part of the public API.

Reply:
Makes scense.